spark/graphar/src/main/java/org/apache/graphar/info/EdgeInfo.java

```diff
@@ -121,9 +122,14 @@ public static EdgeInfo load(String edgeInfoPath, Configuration conf) throws IOEx
         if (conf == null) {
             conf = new Configuration();
         }
-        Path path = new Path(edgeInfoPath);
-        FileSystem fileSystem = path.getFileSystem(conf);
-        FSDataInputStream inputStream = fileSystem.open(path);
+        return load(edgeInfoPath, FileSystem.get(conf));
+    }
+
+    public static EdgeInfo load(String edgeInfoPath, FileSystem fileSystem) throws IOException {
```

Comment:
Maybe the filesystem could be a optional parameter, here it seems that user need to always pass a fileSystem with this method?

Reply:
FileSystem is optional for conveinience, user can also call with hdfs' configuration object.

Reply:
sorry, I don't get the meaning, you mean the fileSystem can be a FileSystem object or Configuration? Is that possible with single function in Java?

Reply:
For the users that are not familiar with Hadoop's FileSystem, I think we should provide a default load function that directly load file from local filesystem. Use `XxxInfo.load(infoPath, null)` is really weird to me, since pass a null we should raise an error, not create a default filesystem in most cases.

Reply:
how about we simply the load function by provide two overload function:
load(String)  # load info from create filesystem with new Configuration();
load(String, Configuration conf) # load info from filesystem create with given configuration

This would give both direct and simple cognition for users that familiar with Hadoop's FileSystem API or not familiar with the API.

Reply:
added